app/src/main/java/ar/com/wolox/android/training/ui/login/LoginPresenter.java

```diff
@@ -68,6 +78,33 @@ public void onTermsAndConditionsClicked() {
         return errors;
     }
 
+    private Callback<List<User>> doLogin(String email) {
+        return new Callback<List<User>>() {
+
+            @Override
+            public void onResponse(@NonNull final Call<List<User>> call, @NonNull final Response<List<User>> response) {
+                final List<User> users = response.body();
+                final boolean isValidUser = users != null && !users.isEmpty() && users.get(0) != null;
```

Comment:
Sorry but I prefer isValidUser to be a method that you ask something and returns a Boolean, instead of creating a new variable to save this value and then test it in the if statement